src/Test/Utilities/DiagnosticAnalyzerTestBase.cs

```diff
@@ -84,6 +88,21 @@ public static MetadataReference SystemThreadingTaskFacadeRef
             }
         }
 
+        protected bool PrintActualDiagnosticsOnFailure
+        {
```

Comment:
`protected bool PrintActualDiagnosticsOnFailure { private get; set; }` sounds good to me. I can do that.

We could make `PrintActualDiagnosticsOnFailure ` with `public get;` too but it doesn't serve any purpose. Tests (or any helpers)are not supposed to read it to determine their execution. So I think keeping it `private get;` is good for now.